src/interfaces/token.spec.ts

```diff
@@ -0,0 +1,475 @@
+        gas: zapperGasMock,
+        supportedTokens: zapperSupportedTokensMock,
+        zapInApprovalState: zapperZapInApprovalStateMock,
+        zapInApprovalTransaction: zapperZapInApprovalTransactionMock,
+        zapOutApprovalState: zapperZapOutApprovalStateMock,
+        zapOutApprovalTransaction: zapperZapOutApprovalTransactionMock
+      }
+    },
+    ironBank: { balances: ironBankBalancesMock },
+    vaults: {
+      balances: vaultsBalancesMock
+    }
+  }))
+}));
+
+jest.mock("../context", () => ({
+  Context: jest.fn().mockImplementation(() => ({
+    provider: {
+      write: {
+        getSigner: jest.fn().mockImplementation(() => ({
+          sendTransaction: jest.fn().mockResolvedValue("transaction")
+        }))
+      }
+    }
+  }))
+}));
+
+describe("TokenInterface", () => {
+  let tokenInterface: TokenInterface<1>;
+
+  let mockedYearn: Yearn<ChainId>;
+
+  beforeEach(() => {
+    mockedYearn = new (Yearn as jest.Mock<Yearn<ChainId>>)();
+    tokenInterface = new TokenInterface(mockedYearn, 1, new Context({}));
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe("price", () => {
+    it("should should get the exchange rate between two tokens", async () => {
+      expect(await tokenInterface.price("0x000", "0x001")).toEqual(1);
+      expect(getPriceFromRouterMock).toHaveBeenCalledTimes(1);
+      expect(getPriceFromRouterMock).toHaveBeenCalledWith("0x000", "0x001");
+    });
+  });
+
+  describe("priceUsdc", () => {
+    it("should get the suggested Usdc exchange rate for a token", async () => {
+      getPriceUsdcMock.mockResolvedValue(0.000001);
+
+      expect(await tokenInterface.priceUsdc("0x000")).toEqual(0.000001);
+      expect(getPriceUsdcMock).toHaveBeenCalledTimes(1);
+      expect(getPriceUsdcMock).toHaveBeenCalledWith("0x000", undefined);
+    });
+
+    it("should get the suggested Usdc exchange rate for list of tokens", async () => {
+      getPriceUsdcMock.mockResolvedValueOnce(0.000001).mockResolvedValueOnce(0.000002);
+
+      expect(await tokenInterface.priceUsdc(["0x000", "0x001"])).toEqual({
+        "0x000": 0.000001,
+        "0x001": 0.000002
+      });
+      expect(getPriceUsdcMock).toHaveBeenCalledTimes(2);
+      expect(getPriceUsdcMock).toHaveBeenNthCalledWith(1, "0x000", undefined);
+      expect(getPriceUsdcMock).toHaveBeenNthCalledWith(2, "0x001", undefined);
+    });
+  });
+
+  describe("balances", () => {
+    const vaultTokenWithBalance = balanceFactory.build({
+      address: "0x001"
+    });
+    const vaultTokenWithoutBalance = balanceFactory.build({
+      balance: "0"
+    });
+    const zapperTokenWithBalance = balanceFactory.build({
+      address: "0x002"
+    });
+    const zapperSameAddressTokenWithBalance = balanceFactory.build({
+      address: "0x001"
+    });
+
+    beforeEach(() => {
+      vaultsBalancesMock.mockResolvedValue([vaultTokenWithBalance, vaultTokenWithoutBalance]);
+    });
+
+    describe("when chainId is 1 or 1337", () => {
```

Comment:
this tests are not actually checking when chainId is 1337. you ould do something like this if we expect the same result for both
```
[1, 1337].forEach((chainId) => 
describe(`when chainId is ${chainId}`, () => {
      beforeEach(() => {
        tokenInterface = new TokenInterface(mockedYearn, chainId, new Context({}));
      });

      it("should fetch token balances from the TokenInterface.supported list", async () => {
        zapperBalancesMock.mockResolvedValue([zapperTokenWithBalance, zapperSameAddressTokenWithBalance]);

        expect(await tokenInterface.balances("0x000")).toEqual([zapperTokenWithBalance, vaultTokenWithBalance]);
        expect(zapperBalancesMock).toHaveBeenCalledTimes(1);
        expect(zapperBalancesMock).toHaveBeenCalledWith("0x000");
        expect(ironBankBalancesMock).not.toHaveBeenCalled();
      });
    });
    )
    ```

Reply:
Good catch! I was just checking the behaviour when it's either of those and totally neglected the fact that I was just testing for `chainId` 1

Addressed with https://github.com/yearn/yearn-sdk/pull/218/commits/1c445501c3baaa6201f0793e166b56b5ede80a26